### DISCUSSION TITLE
FCP 10.8, macOS Sonoma and Apple Silicon Persistent Beachballing Problem when Transcoding

### DISCUSSION THREAD

**Discussion by tangierc** [CommandPost/FCPCafe#391] [Final Cut Pro Questions]
Folks, I keep having an issue while since upgrading to Final Cut Pro 10.8 and macOS Sonoma.

1. I used to have no issues doing additional work after initiating a transcode. FCP would simply pause it's processes.
2. Now whenever I initiate transcoding to make proxies, I get the spinning beachball which persists until the job is done. I can't click anywhere else or do any other operation within Final Cut Pro.
3. I've closed other apps to free up any memory.
4. The background task does continue though.

As of writing this I am using FCP 10.8.1 on macOS Sonoma 14.6.1 on an M2 Max MacBook Pro with 64 GB of memory.
Media is stored on an OWC Gemini RAID and transcoding via thunderbolt to an OWC SSD where all of my project proxies are stored.
This is also after a complete wipe and reinstall of my computer, the macOS, FCP and a clean rebuild of the RAID.

Has anyone else noticed an uptick or more situations where FCP does not allow you to continue working and pause background tasks? I don't recall having this issue with Intel Processors nor before Sonoma.

**Comment by latenitefilms** [comment]
I haven't had any negative reports about 10.8.1 - however, I wonder if previously you had CommandPost's **Enable Rendering During Playback** enabled, so you're used to that behaviour?

**Reply by tangierc** [reply]
Hey Chris. I don't have CommandPost installed at the moment. I have been incrementally adding things back in to see if I notice anything different. This project is nowhere as large as the last one I worked on that you helped me with using BRAW Toolbox. I seem to be getting a lot more spinning beachballs when transcoding or when scrolling event keyword collections, also with less than the last project.

**Reply by latenitefilms** [reply]
So it's not crashing, it's just stalling with a spinning wheel of doom?

You could try **Sample Process** on Final Cut Pro when it's freezing via Activity Monitor and share your results here. It might offer some clues?

**Reply by tangierc** [reply]
Not crashing at all. It continues on with its process of transcoding, but FCP becomes locked up. You can't click the task window scroll, manipulate anything on the timeline, etc. It's just the spinning beachball and essentially you have to wait until it's done. There are some occasions with enough click in and out of the app, that I might be able to do something, but I haven't seen it this bad unless I have a lot going on in the actual timeline which Joe Marler helped look into a while ago ("main UI thread busy doing recursive work (Ozone OZSpline calls in this case"). 

I am still working with OWC to figure out what's up with this new Gemini. This drives cranks hard keeping up with scrolling vertically through event keyword collections or just opening an event via the triangle toggle. It's very unusual behavior I didn't have from my older Gemini RAID. The one I am on is in RAID 0 and seems to perform worse with proxies than the ProRes 4444 Arri clips via thunderbolt 3. It's as if nothing is kept in memory on the drive or compute and it's always searching whenever the user (me) tries to do something in FCP. Drive isn't hardly full either. I digress.

**Reply by latenitefilms** [reply]
To be honest, we stopped using spinning drives at LateNite a few years ago - we only work off SSD's.

**Reply by tangierc** [reply]
Yeah, I hear you and I get that recommendation from others. However this drive was provided by the production team and it (should be) a very good drive that should work without issue with the codecs I am using. I've had better performance on other and older drives, so I am currently left to believe that something has changed either with FCP, Sonoma, or both involving memory allocation / swapping on Apple Silicon; particularly after doing a full rebuild of my computer, the RAID, and pertinent software only. Still more trouble shooting to do. Right now I am working on the project mostly in proxies on an SSD, but truth be told, I shouldn't have to. It's just smaller, more convenient, and more mobile.

**Reply by latenitefilms** [reply]
Are you using OWC SoftRAID or Apple's built-in RAID?

**Reply by latenitefilms** [reply]
Ah right, OWC Gemini has a built-in hardware controller, so ignore me.

**Reply by tangierc** [reply]
This Gemini is hardware RAID (selector on the back), unlike the Thunderbay I was using on the last project that had four drives and was using SoftRAID. I can still use SoftRAID on this Gemini to assess the build of the RAID and RAID status though.
